include/umf/pools/pool_disjoint.h

```diff
@@ -106,6 +106,12 @@ umf_result_t
 umfDisjointPoolParamsSetName(umf_disjoint_pool_params_handle_t hParams,
                              const char *name);
 
+/// @brief Tries to release unused pooled memory back to the provider.
+/// @param pool handle to the memory pool.
+/// @param minSlabsToKeep minimum number of slabs to keep.
+/// @return UMF_RESULT_SUCCESS on success or appropriate error code on failure.
+umf_result_t umfDisjointPoolTrimMemory(void *pool, size_t minSlabsToKeep);
```

Comment:
How this API is going to be used?

Reply:
@vinser52 I definitely wanted to discuss this first. Here, I just wanted to unblock @ldorau in his work and provide the required functionality